experiments/experiments.go

```diff
@@ -270,14 +297,21 @@ func (e *SimpleExperiment) Variant(args map[string]interface{}) (string, error)
 	if !e.targeting.Evaluate(args) {
 		return "", nil
 	}
-	bucketVal, ok := args[e.bucketVal].(string)
-	if !ok {
+	bucketVal, ok := value.(string)
+	if !ok && value != nil {
 		return "", fmt.Errorf(
 			"experiment.SimpleExperiment.Variant: expected bucket val to be a string, actual: %T",
-			args[e.bucketVal],
+			value,
 		)
 	}
 
+	if bucketVal == "" {
+		return "", MissingBucketKeyError{
+			ExperimentName: e.name,
+			ArgsKey:        e.bucketVal,
+		}
+	}
+
```

Comment:
I think all this should happen _before_ the overrides:

https://github.com/reddit/baseplate.py/blob/e557576f5b0c4e618069efe5b11c038b181c3a39/baseplate/lib/experiments/providers/simple_experiment.py#L271-L293